### PR TITLE
Exams

### PR DESCRIPTION
**Pull request by Anyta8989**

### CODE REVIEW

File: Tasks/intersection.js [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-if (attribute_name !== '' && object_2[attribute_name]) {
-object_2[attribute_name] = object_1[attribute_name];   {
-}} else                                                {
-delete object_1[attribute_name];                       {
-                                                       }
-                  }{}
-                  }{}
-return object_1;
+const intersection = (first, second) => {
+    const first_keys = Object.keys(first);
```
Comment (tshemsedinov) [inline Tasks/intersection.js:5]: - Use eslint and prettier
- Respect 2 space indentation

File: Tasks/intersection.js [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
+    const output = {};
+    for (const attribute of first_keys) {
+        if (attribute && first[attribute] === second[attribute])
+            output[attribute] = first[attribute];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:9]:
```suggestion
        if (attribute && first[attribute] === second[attribute]) {
            output[attribute] = first[attribute];
        }
```

File: Tasks/intersection.js [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
+    }
+    return output;
 }
```
Comment (tshemsedinov) [inline Tasks/intersection.js:12]: Semicolon

File: Tasks/validate.js [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
-};
+const isValid = (name) => {
+    if (name === '') return false;
+    if (typeof (name) !== 'string') return false;
+    if (!name.includes(' ')) return false;
+    for (const char of name) {
+        if (char === ' ') continue;
+        const charCode = char.toLowerCase().charCodeAt(0);
+        const aCode = 'a'.charCodeAt(0);
+        const zCode = 'z'.charCodeAt(0);
```
Comment (tshemsedinov) [inline Tasks/validate.js:12]: Optimize loop invariant, use global constants

File: Tasks/intersection.js (hunk 1 above)
Comment (Anyta8989) [reply Tasks/intersection.js]: fixed

File: Tasks/validate.js (hunk 4 above)
Comment (Anyta8989) [reply Tasks/validate.js]: fixed

**Pull request closed without merging**